.github/workflows/release-tagging.yml

```diff
@@ -0,0 +1,53 @@
+name: Release Tagging
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  tagging:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Get version from package.json
+      id: get_version
+      run: |
+        version=$(jq -r '.version // empty' < package.json)
+        if [ -z "$version" ]; then
+          echo "No valid version found in package.json."
+          exit 1
+        fi
+        echo "::set-output name=version::$version"
+
+    - name: Check if tag already exists
+      id: check_tag
+      run: |
+        version=${{ steps.get_version.outputs.version }}
+        if git rev-parse "v$version" >/dev/null 2>&1; then
+          echo "Tag v$version already exists."
+          echo "::set-output name=exists::true"
+          exit 0
+        else
+          echo "::set-output name=exists::false"
+        fi
+      
+    - name: Create and push tag
+      if: steps.check_tag.outputs.exists == 'false'
+      run: |
+        git config --global user.name "github-actions[bot]"
+        git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
Why are the name and email hardcoding?

Reply:
This is the default account for the GitHub Action Bot, we want to use it to distinguish that this release was automatically generated by the bot (Github Action), not by a personal account.

Reply:
Why not use the GitHub Token? What if that account is invalid in the future?